samza-core/src/main/java/org/apache/samza/operators/impl/WindowOperatorImpl.java

```diff
@@ -74,22 +78,35 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(WindowOperatorImpl.class);
 
+  private final WindowOperatorSpec<M, WK, WV> windowOpSpec;
+  private final Clock clock;
   private final WindowInternal<M, WK, WV> window;
   private final KeyValueStore<WindowKey<WK>, WindowState<WV>> store = new InternalInMemoryStore<>();
-  TriggerScheduler<WK> triggerScheduler ;
+  private TriggerScheduler<WK> triggerScheduler;
+
+  // Results to be returned for the current handleMessage and handleTimer call.
+  private List<WindowPane<WK, WV>> currentResults = new ArrayList<>();
```

Comment:
Wondering if it'll be slightly cleaner to remove this, and have individual methods return a `List<WindowPane>` objects that'd fire? 

Perhaps, it's not super-obvious to me, Is there a need to book-keep the current `handleMessage` results? It seems that we are emitting them lazily anyways?  (as opposed to the current implementation that propagates results down the chain eagerly, and mutates the state store later)